### PR TITLE
Wenghe/media stream namespace

### PR DESCRIPTION
**Pull request by WengHe**
## Description
(This PR recovers the PR #1586 , which was reverted because of a project dependencies issue #1621 .)
Added a new sub-namespace: `video.mediaStream` and provided a new API under this namespace: `video.mediaStream.registerForVideoFrame`. The video app should use this new API to get video frames when the capability is supported by the host.

### Main changes in the PR:

1. Added a new API `video.mediaStream.registerForVideoFrame`.
2. Renamed video.VideoFrame to video.VideoFrameData to avoid name collision with the JS native VideoFrame interface.

### Unit Tests added:

Yes

### End-to-end tests added:

No

### Change file added:

Yes

### CODE REVIEW

File: packages/teams-js/package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "test": "jest"
   },
   "dependencies": {
+    "@types/dom-mediacapture-transform": "0.1.2",
```
Comment (TrevorJoelHarris) [inline packages/teams-js/package.json:23]: I assume this is the main difference between this change and the previous version that broke a customer. Before we approve this, you will need to be able to explain to us why this change will fix that customer issue. Also, you will need to be able to reproduce the customer reported issue: https://github.com/OfficeDev/microsoft-teams-library-js/pull/1628

Once you can reproduce their issue, please verify that this fix will no longer break them.

**Comment by TrevorJoelHarris** [comment]
Also, it looks like there are merge conflicts you will need to resolve by hand before this can be checked in.

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]
Please verify that this change will fix the reported customer issue and resolve the merge conflicts.

**Comment by WengHe** [comment]
Before fix:
image

After fix, with the locally built package:
image

File: packages/teams-js/package.json (hunk 1 above)
Comment (WengHe) [reply packages/teams-js/package.json]: Reproduced the issue and confirmed the fix won't bring the same issue again, please see the comment I posted below.

**Comment by TrevorJoelHarris** [comment]
I talked with @WengHe on the side: the issue here was that we were exporting types that were only included as dev dependencies. This works if the test app is inside of the same monorepo (as ours is), but will cause problems once someone externally tries to import the package. We'll have someone look into building new automation to catch this scenario going forward.

**Review by TrevorJoelHarris — APPROVED** [review]

File: packages/teams-js/src/public/video.ts [hunk 2]
```diff
@@ -216,4 +222,159 @@ export namespace video {
...
+    type MediaStreamInfo = {
+      streamId: string;
+    };
+
+    /**
+     * @beta
+     * Register to read the video frames from the media stream provided by the host.
+     * @param frameCallback - The callback to invoke when recieve a video frame from the media stream.
+     * @example
+     * ```typescript
```
Comment (aeunms) [inline packages/teams-js/src/public/video.ts:279]: Nit: Might be good to include comment on what this is commented out/saved for

**Review by aeunms — APPROVED** [review]
🚢

File: packages/teams-js/src/public/video.ts (hunk 2 above)
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/video.ts:279]: This is how you put a code example in the documentation that gets formatted nicely. See here for an example that's already live: https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/search?view=msteams-client-js-latest

**Pull request merged**